src/main/java/jenkins/branch/BranchStatusColumn.java

```diff
@@ -104,4 +106,19 @@ public boolean shownByDefault() {
             return false;
         }
     }
+
+    /**
+     * Hide this column from user views as it would only confuse them.
+     */
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+            return !(descriptor instanceof DescriptorImpl) || context instanceof MultiBranchProjectViewHolder.ViewImpl;
```

Comment:
either it is not `BranchStatusColumn.DescriptorImpl`, in which case we take no action... or it is  `BranchStatusColumn.DescriptorImpl`, in which case we hide it from everywhere except the `MultiBranchProjectViewHolder.ViewImpl` (which is non-user configurable so we could hide it here also, except there may be other things that rewrite and apply the filters, so to be safe we allow it for that non-user configurable view)